### DISCUSSION TITLE
Do workspace-specific pipelines inherit properties from the base pipeline?

### DISCUSSION THREAD

**Discussion by yashsway** [vercel/turborepo#3309] [Help] [answered]
Say I had a turbo config like this:

```json
  "pipeline": {
    "build": {
      "dependsOn": ["^build"],
      "outputs": ["dist/**", "build/**"],
      "env": [
        "SENTRY_AUTH_TOKEN",
        "SENTRY_DSN",
        "SENTRY_PROJECT",
        "SENTRY_ORG",
        "NPM_RC",
        "NPM_TOKEN"
      ]
    },
    "an-app#build": {
      "outputs": [".next/**"],
      "env": ["ENABLE_ROOT_PATH_BUILD_CACHE", "VERCEL_GIT_COMMIT_SHA"]
    },
 }
```

Does the pipeline for `an-app#build` inherit the `env` variables and the `dependsOn` declared in the `build` pipeline ? or will I have to include them all again?
```json
    "an-app#build": {
      "dependsOn": ["^build"],
      "outputs": [".next/**"],
      "env": [
        "ENABLE_ROOT_PATH_BUILD_CACHE", 
        "VERCEL_GIT_COMMIT_SHA", 
        "SENTRY_AUTH_TOKEN",
        "SENTRY_DSN",
        "SENTRY_PROJECT",
        "SENTRY_ORG",
        "NPM_RC",
        "NPM_TOKEN"
       ]
    },
```

The docs (https://turbo.build/repo/docs/core-concepts/monorepos/running-tasks#specific-workspace-tasks) say this:
> Package-tasks do not inherit cache configuration. You must redeclare [outputs](https://turbo.build/repo/docs/reference/configuration#outputs) at the moment.

which makes me feel like they **don't** but the second line says all you have to do is redeclare `outputs`... so does that mean all the other properties _are_ inherited?

**Comment by anthonyshew** [comment, marked as answer]
Package-tasks do not inherit configuration. Package Configurations do. This doc should have all the information you're looking for: https://turbo.build/repo/docs/reference/package-configurations

**Reply by yashsway** [reply]
thank you!!
